### PR TITLE
Messaging component funcitional

### PR DESCRIPTION
**Pull request by Dhruv-97**
Added most of the messaging functionality

### CODE REVIEW

**Comment by sophia-hubscher** [comment]
https://github.com/sknooyen/UBay/assets/33038334/68756863-bd63-4a4f-9adf-6a74a023b579

Loading to the name Buyer1 but to the chat log for this user might be a little confusing. Should we maybe load to a blank chat instead?

File: src/Front-end/Messages.js [hunk 1]
```diff
@@ -1,70 +1,204 @@
...
+//         // clear the form fields
+//         // notify user that their listing was posted
+//       }).catch(error => {
+//         console.error('Error posting data:', error);
+// })}
+
+//sample messages only sent by the current user
+
+//param to pass in through axios, need component to operate with verified email address
+export const USERNAME = "bruh1"
```
Comment (sophia-hubscher) [inline src/Front-end/Messages.js:23]: Should we be doing `const USERNAME = auth.currentUser ? auth.currentUser.email : '';` instead? Also, is there a reason this and USERLIST are exported?

File: src/Front-end/Messages.js [hunk 2]
```diff
@@ -1,70 +1,204 @@
...
+
+export const USERLIST = ["Buyer1", "Seller1", "Buyer2", "Selling2"]
+
+const messages = [
```
Comment (sophia-hubscher) [inline src/Front-end/Messages.js:27]: Can this dummy data be deleted to clean things up or is it still in use?

File: src/Front-end/Messages.js (hunk 1 above)
Comment (Dhruv-97) [reply src/Front-end/Messages.js:23]: We should include the const USERNAME = auth.currentUser ? auth.currentUser.email : ''; , however I need a way to get usernames of those the user wants to start a new chat with. Maybe the message seller button on the listings page can somehow parse the users email?

File: src/Front-end/Messages.js (hunk 2 above)
Comment (Dhruv-97) [reply src/Front-end/Messages.js:27]: It can be removed

File: src/Front-end/Messages.js (hunk 1 above)
Comment (sophia-hubscher) [reply src/Front-end/Messages.js:23]: That would be super doable! If you look at the code for the listing page, we display the seller's ID. That same ID could be passed as a parameter when running the messages page code. Does th

**Review by sophia-y-h — APPROVED** [review]
Looks great! Everything works well on my end

**Pull request merged**